tests/test-derivation.nix

```diff
@@ -1,40 +1,62 @@
-{ pkgs, makeNixvimWithModule, ... }:
+{
+  pkgs,
+  lib ? pkgs.lib,
+  makeNixvimWithModule,
+  ...
+}:
 let
   # Create a nix derivation from a nixvim executable.
   # The build phase simply consists in running the provided nvim binary.
   mkTestDerivationFromNvim =
     {
       name,
-      nvim,
+      nvim ? null,
+      tests ? [ ],
       dontRun ? false,
       ...
     }:
+    assert lib.assertMsg (tests == [ ] || nvim == null) "Both nvim & tests can't be supplied";
```

Comment:
I think this should be `&&`?

Also need to assert that at least one is defined:

```nix
assert lib.assertMsg (tests != [ ] || nvim != null) "Either `nvim` or `tests` must be provided.";
```

---

Would it be easier to use `@args` and then do `args ? tests && args ? nvim` (etc)?

Reply:
It's not a problem that none are defined, it's only a problem if both are defined

Reply:
> It's not a problem that none are defined, it's only a problem if both are defined

:+1: ok

> I think this should be `&&`?

I think this is still the case though? Unless I'm missing some logic?

Reply:
It should be fine, because the assert will trigger if `tests == [ ] || nvim == null` evaluates to `false`. This means that `tests != [ ]` and `nvim != null`, which is the error we are trying to detect.
(asserts are very confusing to check for error paths though --')

Reply:
Yes, sorry I was getting mixed up! Maybe let's add a clarifying comment?

Reply:
Done!